problem-gson/src/main/java/org/zalando/problem/StatusTypeAdapter.java

```diff
@@ -0,0 +1,43 @@
+package org.zalando.problem;
+
+import com.google.gson.TypeAdapter;
+import com.google.gson.stream.JsonReader;
+import com.google.gson.stream.JsonToken;
+import com.google.gson.stream.JsonWriter;
+
+import javax.annotation.Nullable;
+import java.io.IOException;
+import java.util.Map;
+import java.util.Objects;
+
+final class StatusTypeAdapter extends TypeAdapter<StatusType> {
+
+    private Map<Integer, StatusType> index;
+
+    StatusTypeAdapter(Map<Integer, StatusType> statuses) {
+        this.index = statuses;
+    }
+
+    @Override
+    public void write(JsonWriter out, @Nullable StatusType status) throws IOException {
+        if (Objects.isNull(status)) {
+            out.nullValue();
```

Comment:
Ideally the whole key would be absent:

```json
{"type":"test"}
```

rather than

```json
{"type":"test","status":null}
```

Reply:
Gson is by default configured to ignore the `null` fields. So the **status** field would be absent in the output. 

In certain scenarios, one might want to serialize null fields also, hence this handling.
Also, to specify, following is the way to enable null field serialization.
```java
Gson gson = new GsonBuilder().serializeNulls().create();
```

Reply:
Good to know :+1: